### ISSUE TITLE
Framerate independent update() calls with delta_time

### ISSUE THREAD

**Issue by desdesmo** [pvcraven/arcade#266]
## Bug Report
spritelist & sprite are using only the function parameter "self"
a typical framerate independent update() function should use delta_time to work properly.

### Actual behavior:
Framerate dependent update() of sprite movement

### Expected behavior:
Framerate independent update() of sprite movement

Example "Move a Sprite with animation"
http://arcade.academy/examples/sprite_move_animation.html?highlight=sprite%20update

Code snippet from row 153:
def update(self, delta_time):
        """ Movement and game logic """
        self.all_sprites_list.update()

Should be passing the delta_time on to subsequent calls
def update(self, delta_time):
        """ Movement and game logic """
        self.all_sprites_list.update(delta_time)

Thus the classes spritelist and sprite should implement the delta_time.
Otherise everybody needs to overwrite the functions to enable this normal feature.

**Comment by pvcraven** [comment]
Ideally, yes, this is how it should be. Although adding this would be a breaking API change that would invalidate much of the current code. I'm open to a solution if someone has a good idea.

**Comment by desdesmo** [comment]
maybe with a faked method overloading? method overloading is not a feature of python, but with a default var might be achived.

def update(self, dt=None):
        if i is None:
            #old_code
        else:
            #new_code using delta_time

**Comment by pvcraven** [comment]
I'm thinking of it from the other side.

For example:
http://cs.simpson.edu/schedule/class_schedule.php

Line 387, if I updated the `update` function I'd need to pass in `dt`:
```for sprite in self.sprite_list:
    sprite.update()```

But if I run this against a programmer's class uses old code, it won't take a `dt`. An exception will be caused breaking the backward compatibility of the code.

**Comment by desdesmo** [comment]
i think a default typed parameter is not mandatory when calling the method..
so even a call with the missing dt will work.

for reference:  https://stackoverflow.com/questions/10202938/how-do-i-use-method-overloading-in-python

**Comment by dangillet** [comment]
It’s unfortunately not allowed to call a function with additional keyword arguments. What you described is when you want the client code to be able to call you with or without dt. But here, we’re talking about arcade calling client update method with an additional dt parameter. 

Older code would break as they have overloaded update with the old signature. 

Now I do agree all update calls should include dt. What about fixing this for version 2?

**Comment by pvcraven** [comment]
Right, I struggle with that a bit. Most of v2 is backwards compatible. But this would break pretty much all the programs.

**Comment by pauleveritt** [comment]
Perhaps a different flavor of "update" which has the new signature.

I have a feeling we are, at some point, going to have the big conversation about "is Arcade the pyglet UI or not"? If we decide not, we could resurrect the idea of a new "Game API", where we do the calling not pyglet.

**Comment by desdesmo** [comment]
ok, got your point. after thinking about it again, i would suggest a 2nd method with a custom signature like update_dt(dt).

another thought, maybe you can share your opinion:
if i am not willing to change the signature in custom classes i could use a global var that recieves the delta_time once per update. at the very beginning of the calls. other methods could use the global_var.

it reduces the passing of the dt as a parameter (and reducing possibly overhead/memory in subsequent calls). of course a global should always be used carefully. in this case, where DT is stable/constant for each call, it might be worth a thought.

**Comment by pvcraven** [comment]
To be included in 2.2.6 through `on_update` function.

**Comment by pvcraven** [comment]
Fixed with release 2.2.6.
